src/backend/marsha/core/api.py

```diff
@@ -150,6 +211,125 @@ def initate_upload(self, request, pk=None):
 
         return Response(presigned_post)
 
+    @action(methods=["post"], detail=True, url_path="initiate-live")
+    # pylint: disable=unused-argument
+    def initiate_live(self, request, pk=None):
+        """Create a live stack on AWS ready to stream.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+        pk: string
+            The primary key of the video
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse with the serialized video.
+        """
+        now = timezone.now()
+        stamp = to_timestamp(now)
+
+        video = self.get_object()
+        key = f"{video.pk}_{stamp}"
+
+        live_info = create_live_stream(key)
+
+        Video.objects.filter(pk=pk).update(
+            live_info=live_info,
+            upload_state=defaults.LIVE,
```

Comment:
I did that at the beginning but it was more complicated to manage it this way. Because the `upload_state` field will be in an "weird" state. I prefer to set it with `live`, it's explicit.

Reply:
I have to disagreed. It's not explicit at all that the "upload_state" is "live" as the live functionality has nothing to do with uploading :man_shrugging: I think it's ok that the "upload_state" remains "pending" when doing a live... It's the default state and yes it's true that upload is pending as nothing was uploaded yet...

Reply:
Done